mpt_trie/src/trie_ops.rs

```diff
@@ -441,36 +477,35 @@ fn insert_into_trie_rec<N: PartialTrie>(
                 value.clone(),
             );
 
-            Some(place_branch_and_potentially_ext_prefix(
+            Ok(Some(place_branch_and_potentially_ext_prefix(
                 &info,
                 existing_node_truncated,
                 new_node,
-            ))
+            )))
         }
     }
 }
 
 fn delete_intern<N: PartialTrie>(
     node: &Node<N>,
     mut curr_k: Nibbles,
-) -> Option<(WrappedNode<N>, Vec<u8>)> {
+) -> TrieOpResult<Option<(WrappedNode<N>, Vec<u8>)>> {
     match node {
         Node::Empty => {
             trace!("Delete traversed Empty");
-            None
+            Ok(None)
         }
-        Node::Hash(_) => {
-            panic!("Attempted to delete a value that ended up inside a hash node")
-        } // TODO: Find a nice way to get the full key path...
+        Node::Hash(_) => Err(TrieOpError::HashNodeDeleteError()),
+        // TODO: Find a nice way to get the full key path...
         Node::Branch { children, value } => {
             if curr_k.is_empty() {
-                return Some((branch(children.clone(), Vec::new()), value.clone()));
+                return Ok(Some((branch(children.clone(), Vec::new()), value.clone())));
             }
 
             let nibble = curr_k.pop_next_nibble_front();
             trace!("Delete traversed Branch nibble {:x}", nibble);
 
-            delete_intern(&children[nibble as usize], curr_k).map(
+            delete_intern(&children[nibble as usize], curr_k)?.map_or(Ok(None),
                 |(updated_child, value_deleted)| {
                     // If the child we recursively called is deleted, then we may need to reduce
                     // this branch to an extension/leaf.
```

Comment:
Small nit, but wdyt about instead replacing the `map_or` with `map` and then wrapping `delete_intern` with an `Ok()`? I guess this is a bit subjective, but this mean we can ignore the `Result` type except for the `Ok()` that wraps everything.

Reply:
but we have:
```
updated_children[nibble as usize] =
                                try_collapse_if_extension(updated_child)?;
 ```
 inside the map_or